src/raftstore/store/peer.rs

```diff
@@ -484,6 +494,34 @@ impl Peer {
         StaleState::Valid
     }
 
+    fn next_lease_expired_time(&self, send_to_quorum_ts: Timespec) -> Timespec {
+        // The valid leader lease should be
+        // "lease = election_timeout - (quorum_commit_ts - send_to_quorum_ts)"
+        // And the expired timestamp for that leader lease is "quorum_commit_ts + lease",
+        // which is "send_to_quorum_ts + election_timeout" in short.
+        send_to_quorum_ts + self.election_timeout
+    }
+
+    fn update_leader_lease(&mut self, ready: &Ready) {
+        // Update leader lease variables when the raft state changes
+        if ready.ss.is_some() {
+            let ss = ready.ss.as_ref().unwrap();
+            match ss.raft_state {
+                StateRole::Leader => {
+                    self.leader_lease_expired_time =
```

Comment:
> Do you mean those processing inside of raft library

No, still in peer, not in raft. 

I proposed a simple solution before.
We can use t2 - t1 < 1s to do more safe check. 
t1 is the start time when proposing, and t2 is the end time when applying, if the total time is less than the boundary, we can think the leader is still the leader. 

Here and following leader_lease_expired_time update is not safe. Let's assume the following case:

1. The peer sends votes at t1
2. All peers votes for it
3. The peer becomes leader at t2, but call on_ready at t3

If t3 - t2 > election timeout, the leader may be stale. So we can't think [t3, t3 + 5s) is safe for lease read.